torch/csrc/jit/autodiff.cpp

```diff
@@ -350,7 +358,21 @@ static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_val
         auto grad = grads.at(0);
         std::vector<SymbolicVariable> tensor_grads;
         for (auto input : tensor_inputs) {
-          tensor_grads.push_back(grad.narrow(dim, offset, input.sizes()[dim]));
+          if (pad) {
+            int64_t ndim = input.sizes().size();
+            for (int64_t i = 0; i < ndim; ++i) {
+              auto result = grad;
+              if (i == dim) {
```

Comment:
This doesn't look right...
Let's say I have tensors x, y of size (1, 2, 3) and (1, 3, 2) and I'm doing `torch.cat([x, y], dim=0, pad_value=0`.
The result has size (2, 3, 3).

Now, you're given a grad of size (2, 3, 3). You have to slice out (1, 2, 3) and (1, 3, 2) from this, so you get that by:
```
grad_x = grad.narrow(dim, 0, 1).narrow(...).narrow(...)  # need to narrow along dim 1 and dim 2 s.t. the sizes match
grad_y = grad.narrow(dim, 1, 1).narrow(...).narrow(...) # need to narrow along dim 1 and dim 2 s.t. the sizes match